### PR TITLE
Add world map overlay to clue scroll plugin

### LINKED ISSUE #1446: Show map for coordinate and map clues
It would be nice if Clue scrolls that ask for coordinates had a map icon that you could press which would show you exactly where to go, this would save players a bit of time going on google and typing the coordinates them selves. 

Example of coordinates 

> 00 degrees 05 minutes south, 01 degrees 13 minutes east

or an alternative that could be that it could say the clue destination instead of the regular 

> Coordinate clue | Travel to the marked out destination to see a marker for where you should dig.

### PR DESCRIPTION
**Pull request by MESLewis**
Add overlays that are drawn on the world map, correctly handling zoom, and the resizing of the world map.

Also adds a WorldMapService to aid with further additions to the world map and general world map usage by plugins.

Setting the map location and adding an overlay dot there (Developer plugin: WorldMapOverlayTest):
smoothjump

Drawing image on map, with snapping to map edges:
movesnap1

Clickable overlays:
overlayclick

Will fix #1446 

Todo
- [x] Fix coordinates detection
- [x] Remove Plugin param from WorldMapPoint
- [x] Make WorldMapPoint more flexible
- [x] Change clue image when snapped to edge to be correctly visible on top edge
- [X] Use Mixin to augment functionality
- [X] Move from overlay to manager
- [X] Clip rendering to map bounds
- [X] Smooth map movement. Mimic how search does it.
- [X] Add clue scroll overlay displaying an image at the clue location
- [X] Overlay click to center map
- [X] Optionally snap overlays to the edge of map (like the 'you are here' arrows)

### CODE REVIEW

**Comment by MagicfTail** [comment]
I know this is still very much wip, but I think we should focus on making the map less terrible to use before adding more features to it. It freezes the game when opening, kills your performance, feels clunky/choppy when moving, doesn't close with escape, and the list goes on. That doesn't mean managing to add stuff to this bowl of spaghetti Jagex calls a map isn't impressive, I just don't feel like it should be the focus at the moment.

**Comment by MESLewis** [comment]
One of my long term plans is to look at improving map performance through a custom map region renderer. But that is a ways off. The main goal of this pull is to give us clue markers, that is a feature that we have been wanting for weeks and I want to focus on that first

**Comment by MagicfTail** [comment]
But if that's the case, I would prefer using Death's, although limited, very fast clue marker than having my client freeze every other clue step. No matter how good the clue markers themselves are, it will be a pain to use if the map they are built upon is terrible.

**Comment by MESLewis** [comment]
This doesn't stop anything that deaths PR is doing. Last I heard there were a few things to get sorted out related to downloads before anything can use the dumped map images.

**Comment by MagicfTail** [comment]
I know, and as I said it is impressive that you can add stuff to that pos map. I'm just saying the clue helper won't really be usable no matter how good you make it unless the map itself is improved, and that it's, therefore, *my* opinion that the focus should be on improving the map before adding clue stuff to it. It's ofc up to you what you want to focus on.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+	//Returns in world coordinates
+	public int getWorldMapY()
+	{
+		return surfaceToWorldY(client.getRenderOverview().getWorldMapY());
+	}
+
+
+	//Surface is the native map, world is RuneLites
+	private int surfaceToWorldX(int surfaceX)
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java]: Make it take in point and do vector math

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	{
+		return client.getRenderOverview().getWorldMapDisplayHeight();
+	}
+
+	private int getDisplayX()
+	{
+		return client.getRenderOverview().getWorldMapDisplayX();
+	}
+
+	private int getDisplayY()
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java]: Change there to points

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+			}
+		}
+		else
+		{
+			graphics.setClip(worldMap.getMapDisplayRectangle());
+		}
+
+		Dimension dimension = this.render(graphics, drawPoint);
+		this.renderRegion.setSize(dimension);
+		this.renderRegion.setLocation((int) (drawPoint.getX() - dimension.getWidth() / 2), (int) (drawPoint.getY() - dimension.getHeight() / 2));
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java]: Will calculate incorrectly for non centered drawings. Make this render return a Rectangle instead of Dimension to fix this issue easily and cleanly.

**Comment by SoyChai** [comment]
Potential marker arrow that @sethtroll put together, based on OSB's. I did a quick edit to help it match the pixel style of other map elements.

image image

arrow4x arrowpx4x

**Comment by MESLewis** [comment]
I'm thinking of taking that arrow outline and adding emote / spade / scroll like is done with the ground overlay for clues.

**Comment by MESLewis** [comment]
Emote and spade didnt fit as cleanly in the arrow background so those aren't included in this request.

**Comment by deathbeam** [comment]
I don't think this should be overlay based. I mentioned this before, but the way this is implemented is weird, it should be done in similar way as tooltip manager and tooltips do it.

**Comment by MESLewis** [comment]
Can you elaborate? Having it be overlay based creates a solid foundation and let's plugins use the world map easily and in a customizable manner.

I'm not familiar with how the tooltips are done. The world map should support multiple overlays at once for future functionality.

**Comment by deathbeam** [comment]
The points on minimap should represent data, not renderable entities. Also, each of them has mouse listeners. You need only 1 mouse listener as you will be clicking 1 thing at once. So, the points should be represented something like this:

```java
@Value
@Builder
public class WorldMapPoint
{
  private final Point position;
  private String tooltip;
  private BufferedImage image;
  private Runnable onClick;
}
```

and then make something like WorldMapPointManager that will have list of this points, render them all, handle their onClicks and also have something like setActivePoint.

**Comment by MESLewis** [comment]
There is a single mouse listener that calls the overlay on click when needed.

The issue with a simple data representation is the world map needs to support more complex drawing in the future, for instance drawing friend locations could have a dot and their name above it.

**Comment by MESLewis** [comment]
It's possible to not use overlays but I don't see the disadvantage using an established system instead of creating another one

**Comment by deathbeam** [comment]
I disagree. With flexible enough data structure you could have everything you would ever need. Your implementation is reliant on injecting bunch of stuff to bunch of overlays, and will be really annoying to work with with large amount of points at once. And, as I said earlier, the world map points are plain data, representing them as something else is incorrect imo.

The established system is not designed to work with hundreds of overlays. But, you can have hunderds of world points at once. Also, it is not designed to work with dynamic number of overlays, what you would need with world map points.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	@Inject
+	Client client;
+
+	@Inject
+	public WorldMapService(Client client)
+	{
+		this.client = client;
+	}
+
+	public WorldPoint getMapCenter()
```
[the remark this answers is not in the document]
Comment (MESLewis) [reply runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java]: I did it this way to seperate the jagex logic from our logic. They use offset coordinates internally in every case I've seen except 1. I can move it if you want but I don't know how to organize it in a way that is better than what is there now.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+		Widget widget = client.getWidget(WORLD_MAP, 0);
+		if (widget == null)
+		{
+			return null;
+		}
+
+		for (WorldMapPoint worldPoint : points)
+		{
+			if (worldPoint.getImage() != null && worldPoint.getWorldPosition() != null)
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java]: Inverse if, then it reads better

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+				{
+					drawPoint = clipToRectangle(drawPoint, worldMapRectangle);
+				}
+				else
+				{
+					graphics.setClip(worldMapRectangle);
+				}
+
+				graphics.drawImage(worldPoint.getImage(), drawPoint.getX() - worldPoint.getImagePoint().getX(), drawPoint.getY() - worldPoint.getImagePoint().getY(), null);
+				Rectangle clickbox = new Rectangle(drawPoint.getX() - worldPoint.getImagePoint().getX(), drawPoint.getY() - worldPoint.getImagePoint().getY(), worldPoint.getImage().getWidth(), worldPoint.getImage().getHeight());
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java]: there should really be some code style restriction on max width :/

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlayMouseListener.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+		if (e.getButton() == MouseEvent.BUTTON1)
+		{
+			Point mousePos = clientProvider.get().getMouseCanvasPosition();
+
+			for (WorldMapPoint worldMapPoint : worldMapPointManager.getWorldMapPoints())
+			{
+				if (worldMapPoint.getClickbox().contains(mousePos.getX(), mousePos.getY()))
+				{
+					if (worldMapPoint.getOnClick() != null)
+					{
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlayMouseListener.java]: Invert these if's

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlayMouseListener.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+						worldMapPoint.getOnClick().run();
+						e.consume();
+						continue;
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlayMouseListener.java]: I don't see any reason for `continue` here

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapPoint.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import java.awt.Rectangle;
+import java.awt.image.BufferedImage;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import net.runelite.api.Point;
+import net.runelite.api.coords.WorldPoint;
+
+@Data
+@AllArgsConstructor
+public class WorldMapPoint
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapPoint.java]: As far as I have seen this can be immutable. Add `@Builder` and use it. It will be much nicer then

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+	private int getSurfaceOffsetY()
+	{
+		return clientProvider.get().getRenderOverview().getWorldMapManager().getSurfaceOffsetY();
+	}
+
+	private float getSurfaceZoom()
+	{
+		return clientProvider.get().getRenderOverview().getWorldMapZoom();
+	}
+
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java]: Extra space

**Comment by tr33s** [comment]
I tested it, but the coordinate clues aren't getting set it seems like.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java (hunk 5 above)
Comment (SoyChai) [reply runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java]: ❔ I don't see why, this is the normal pattern for a double null-check in Java code.

**Comment by MESLewis** [comment]
@tr33s I am unable to reproduce the coordinate clue issue. I will do further testing and I ask any other developers doing clues to do the same

File: runelite-api/src/main/java/net/runelite/api/RenderOverview.java [hunk 11]
```diff
@@ -24,8 +24,19 @@
...
 public interface RenderOverview
 {
+	Point getWorldMapPosition();
+
+	Rectangle getWorldMapDisplay();
+
+	float getWorldMapZoom();
+
+	//Takes a world point
+	void setWorldMapPositionTarget(int worldPointX, int worldPointY);
```
Comment (Adam-) [inline runelite-api/src/main/java/net/runelite/api/RenderOverview.java]: Should this actually take a `WorldPoint`?

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.overlay.worldMap;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java]: Don't camelCase packages

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import net.runelite.api.Client;
+import net.runelite.api.Point;
+import net.runelite.api.RenderOverview;
+import net.runelite.api.coords.WorldPoint;
+
+@Singleton
+public class WorldMapService
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapService.java]: I would move all of these functions to a mixin, except for `mapWorldPointToGraphicsPoint` which goes into the overlay.

File: runelite-api/src/main/java/net/runelite/api/RenderOverview.java (hunk 11 above)
Comment (MESLewis) [reply runelite-api/src/main/java/net/runelite/api/RenderOverview.java]: The comment is referring to the parameters being in world coordinates instead of worldMap coordinates. Most functions in RenderOverview take worldMap coordinates

Comment (MESLewis) [reply runelite-api/src/main/java/net/runelite/api/RenderOverview.java]: Since the other functions taking worldMap coordinates aren't exposed anymore I'll remove the comment

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapPoint.java [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+import java.awt.Rectangle;
+import java.awt.event.MouseEvent;
+import java.awt.image.BufferedImage;
+import lombok.Getter;
+import lombok.Setter;
+import net.runelite.api.Point;
+import net.runelite.api.coords.WorldPoint;
+
+public class WorldMapPoint
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapPoint.java]: You can just annotate this class with `@Data`

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmaptest/WorldMapOverlayTestPlugin.java [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+import net.runelite.api.events.CommandExecuted;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPoint;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPointManager;
+
+@PluginDescriptor(
+	name = "WorldMapOverlayTest",
+	developerPlugin = true
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmaptest/WorldMapOverlayTestPlugin.java]: I would also make this plugin disabled by default

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+	private static final int TOOLTIP_OFFSET_WIDTH = 5;
+	private static final int TOOLTIP_PADDING_HEIGHT = 1;
+	private static final int TOOLTIP_PADDING_WIDTH = 2;
+
+	private final WorldMapPointManager worldMapPointManager;
+	private final Provider<Client> clientProvider;
+	private final WorldMapOverlayMouseListener worldMapOverlayMouseListener;
+
+	@Inject
+	public WorldMapOverlay(Provider<Client> clientProvider, WorldMapPointManager worldMapPointManager, WorldMapOverlayMouseListener worldMapOverlayMouseListener)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java]: Make this constructor private

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+	{
+		this.clientProvider = clientProvider;
+		this.worldMapPointManager = worldMapPointManager;
+		this.worldMapOverlayMouseListener = worldMapOverlayMouseListener;
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.HIGH);
+		setLayer(OverlayLayer.ALWAYS_ON_TOP);
+	}
+
+	public Point mapWorldPointToGraphicsPoint(WorldPoint worldPoint)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java]: Why is this method public? It looks unused

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+
+@Singleton
+public class WorldMapOverlayMouseListener extends MouseListener
+{
+	private final Provider<Client> clientProvider;
+	private final MouseManager mouseManager;
+	private final WorldMapPointManager worldMapPointManager;
+
+	@Inject
+	public WorldMapOverlayMouseListener(Provider<Client> clientProvider, MouseManager mouseManager, WorldMapPointManager worldMapPointManager)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java]: Make this constructor private

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+	{
+		this.clientProvider = clientProvider;
+		this.mouseManager = mouseManager;
+		this.worldMapPointManager = worldMapPointManager;
+		mouseManager.registerMouseListener(0, this);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java]: Why this needs to be the first one in the list?

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	@Override
+	public MouseEvent mousePressed(MouseEvent e)
+	{
+		if (e.getButton() == MouseEvent.BUTTON1)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java]: Use SwingUtillities.isLeftMouseButton (or something like that, not 100% sure about the method name now)

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+					&& worldMapPoint.getClickbox().contains(mousePos.getX(), mousePos.getY()))
+				{
+					return worldMapPoint.onClick(e);
+				}
+			}
+		}
+		return e;
+	}
+
+	private WorldMapPoint tooltipPoint = null;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java]: Move this declaration next to other field declarations

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapPointManager.java [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Collection;
+import java.util.List;
+import javax.inject.Singleton;
+import lombok.Getter;
+
+@Singleton
+public class WorldMapPointManager
+{
+	@Getter
+	private final List<WorldMapPoint> worldMapPoints = new ArrayList<>();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapPointManager.java]: Make this getter package-private

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java [hunk 23]
```diff
@@ -426,7 +475,7 @@ private CoordinateClue coordinatesToWorldPoint(String text)
 			return null;
 		}
 
-		if (!splitText[1].startsWith("degree") || !splitText[3].startsWith("minute"))
+		if (!splitText[1].equals("degrees") || !splitText[3].equals("minutes"))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: Why is this changed? looks like incorrect rebase to me

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+package net.runelite.client.plugins.cluescrolls;
+
+import java.awt.Graphics;
+import java.awt.event.MouseEvent;
+import java.awt.image.BufferedImage;
+import net.runelite.api.Client;
+import net.runelite.api.Point;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPoint;
+
+public class ClueScrollWorldMapPoint extends WorldMapPoint
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java]: Make this package protected, same with constructor

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java [hunk 25]
```diff
@@ -234,9 +278,13 @@ public void onGameTick(final GameTick event)
...
 
 				// Set hint arrow to first NPC found as there can only be 1 hint arrow
-				if (config.displayHintArrows() && npcsToMark.length >= 1)
+				if (npcsToMark.length >= 1)
 				{
-					client.setHintArrow(npcsToMark[0]);
+					if (config.displayHintArrows())
+					{
+						client.setHintArrow(npcsToMark[0]);
+					}
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: new line after } please

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java [hunk 26]
```diff
@@ -222,6 +265,7 @@ public void onGameTick(final GameTick event)
 			{
 				client.setHintArrow(location);
 			}
+			worldMapPoint.setWorldPoint(location);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: Empty line after }

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java [hunk 27]
```diff
@@ -87,6 +91,10 @@
 public class ClueScrollPlugin extends Plugin
 {
 	private static final Duration WAIT_DURATION = Duration.ofMinutes(4);
+	public static final BufferedImage EMOTE_IMAGE;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: This looks, hmm, unused? Why is it here

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java [hunk 28]
```diff
@@ -0,0 +1,246 @@
...
+				if (worldPoint.getImagePoint() == null)
+				{
+					drawX -= image.getWidth() / 2;
+					drawY -= image.getHeight() / 2;
+				}
+				else
+				{
+					drawX -= worldPoint.getImagePoint().getX();
+					drawY -= worldPoint.getImagePoint().getY();
+
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java]: extra whitespace

**Comment by xpepi** [comment]
add instance map for cord clue scroll like in https://github.com/runelite/runelite/pull/1373

**Comment by MESLewis** [comment]
This PR is closed. You are welcome to submit a new PR with your proposed changes
